hail/src/main/scala/is/hail/expr/ir/Simplify.scala

```diff
@@ -141,6 +141,8 @@ object Simplify {
 
     case x@ArrayMap(NA(_), _, _) => NA(x.typ)
 
+    case ArrayZip(as, names, body, _) if as.length == 1 => ArrayMap(as.head, names.head, body)
```

Comment:
You might also do
```scala
case ArrayMap(ArrayZip(as, names, body1, b), name, body2) =>
  ArrayZip(as, names, Let(name, body1, body2), b)
```
to allow optimizations across `body1` and `body2`.

If one of the `as` is an `ArrayZip`, we can fuse them, but I'm not sure how to write that.

Reply:
good call. Fusing zips doesn't seem hard but will leave that for another PR.